_observing-your-data/alerting/comp-monitors.md

```diff
@@ -0,0 +1,633 @@
+    "inputs": [
+        {
+            "composite_input": {
+                "sequence": {
+                    "delegates": [
+                        {
+                            "order": 1,
+                            "monitor_id": "grsbCIcBvEHfkjWFeCqb"
+                        },
+                        {
+                            "order": 2,
+                            "monitor_id": "agasbCIcBvEHfkjWFeCqa"
+                        }
+                    ]
+                }
+            }
+        }
+    ],
+    "enabled_time": 1679468231835,
+    "enabled": true,
+    "workflow_type": "composite",
+    "schema_version": 0,
+    "name": "scale_up",
+    "triggers": [
+        {
+            "chained_alert_trigger": {
+                "id": "m1ANDm2",
+                "name": "jnkjn",
+                "severity": "1",
+                "condition": {
+                    "script": {
+                        "source": "(monitor[id={{m1}}] && monitor[id={{m2}}])", //ALERT WILL BE CREATED IF MONITOR 1 GENERATES ALERTS AND MONITOR 2 DOESN'T GENERATE ALERT",
+                        "lang": "painless"
+                    }
+                }
+            }
+        },
+        {
+            "chained_alert_trigger": {
+                "id": "m1ORm2",
+                "name": "jnkjn",
+                "severity": "1",
+                "condition": {
+                    "script": {
+                        "source": "(monitor[id={{m1}}] || monitor[id={{m2}}])", //ALERT WILL BE CREATED IF MONITOR 1 GENERATES ALERTS OR MONITOR 2 GENERATE ALERT",
+                        "lang": "painless"
+                    }
+                }
+            }
+        }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+#### Using Painless scripting language to define alert trigger chains
+
+Composite monitor configuration employs the [Painless scripting language](https://www.elastic.co/guide/en/elasticsearch/painless/7.17/painless-guide.html) to define the conditions for chaining alerts. Conditions are applied for each execution of the composite monitor. You specify the definition for the chain in the `triggers.chained_alert_triggers.condition.script.source` field of the request. Using Painless syntax, you can apply logic to links between monitors with basic Boolean operators AND, OR, NOT, and precedence.
+
+* AND = `&&`
+* OR = `||`
+* NOT = `!`
+* Precedence = `()`
+
+See the following examples to understand how each is used in the monitor definition.
+
+* **Example 1**
+   
+   `monitor[id=1] && monitor[id=2]`
+   
+   The following conditions for delegate monitors will trigger the composite monitor to produce an alert when both monitor #1 AND monitor #2 generate an alert.
+
+* **Example 2**
+   
+   `monitor[id=1] || monitor[id=2]`
+
+   The following conditions will trigger the composite monitor to produce an alert when either monitor #1 OR monitor #2 generates an alert.
+
+* **Example 3**
+   
+   `monitor[id=1] && (!monitor[id=2] || monitor[id=3])`
```

Comment:
lets change the example monitor[id=1] && (monitor[id=2] || monitor[id=3]) and corresponding explanation of this syntax